backend/dataall/db/api/target_type.py

```diff
@@ -1,64 +1,43 @@
 import logging
 
-from .. import exceptions, permissions
-from .. import models
+from dataall.db import exceptions, permissions
 
 logger = logging.getLogger(__name__)
 
 
 class TargetType:
-    @staticmethod
-    def get_target_type_permissions():
-        return dict(
-            dataset=(permissions.GET_DATASET, permissions.UPDATE_DATASET),
-            environment=(permissions.GET_ENVIRONMENT, permissions.UPDATE_ENVIRONMENT),
-            notebook=(permissions.GET_NOTEBOOK, permissions.UPDATE_NOTEBOOK),
-            mlstudio=(
-                permissions.GET_SGMSTUDIO_NOTEBOOK,
-                permissions.UPDATE_SGMSTUDIO_NOTEBOOK,
-            ),
-            pipeline=(permissions.GET_PIPELINE, permissions.UPDATE_PIPELINE),
-            redshift=(
-                permissions.GET_REDSHIFT_CLUSTER,
-                permissions.GET_REDSHIFT_CLUSTER,
-            ),
-        )
+    """Resolves the read/write permissions for different type of resources (target types)"""
+    _TARGET_TYPES = {}
+
+    def __init__(self, name, read_permission, write_permission):
+        self.name = name
+        self.read_permission = read_permission
+        self.write_permission = write_permission
+
+        TargetType._TARGET_TYPES[name] = self
```

Comment:
[Clarification] - Smart way of creating the dic. Everytime we define a TargetType we are registering it in _TARGET_TYPES. With this we can avoid hardcoding the supported_target_types
I like using the read and write permissions instead of relying on indexes